model-optimizer/extensions/front/tf/NonConstBeginStridedSliceReplacement.py

```diff
@@ -0,0 +1,136 @@
+                   ('end', dict(op='Pack')),
+                   ('step', dict(op='Const')),
+                   ('strided_slice', dict(op='StridedSlice')),
+                   ],
+            edges=[('begin', 'strided_slice', {'in': 1}),
+                   ('end', 'strided_slice', {'in': 2}),
+                   ('step', 'strided_slice', {'in': 3})])
+
+    def replace_sub_graph(self, graph: Graph, match: dict):
+        strided_slice_node = match['strided_slice']
+        begin_node = match['begin']
+        end_node = match['end']
+        step_node = match['step']
+
+        # retrieve attribute values
+        begin_mask = strided_slice_node.soft_get('begin_mask')
+        end_mask = strided_slice_node.soft_get('end_mask')
+        shrink_mask = strided_slice_node.soft_get('shrink_axis_mask', int64_array([0]))
+
+        # check applicability of this transformation to the given sub-graph:
+        # 1. check that slicing is performed along just one axis
+        if np.sum(begin_mask) != 1 or np.sum(end_mask) != 1 or np.sum(shrink_mask) != 1:
+            return
+        # 2. check that shrink axis is equal to slicing axis
+        if not np.array_equal(np.argwhere(begin_mask == 1), np.argwhere(end_mask == 1)) or \
+                not np.array_equal(np.argwhere(begin_mask == 1), np.argwhere(shrink_mask == 1)):
+            return
+        sliced_axis = np.argwhere(begin_mask == 1)[0][0]
+        # 3. check constant nodes for begin and end correspond to non-slicing axes
+        for idx_port, in_port in begin_node.in_ports().items():
+            if idx_port != sliced_axis and in_port.get_source().node.soft_get('type') != 'Const' or \
+                    idx_port == sliced_axis and in_port.get_source().node.soft_get('type') == 'Const':
+                return
+        for idx_port, in_port in end_node.in_ports().items():
+            if idx_port != sliced_axis and in_port.get_source().node.soft_get('type') != 'Const' or \
+                    idx_port == sliced_axis and in_port.get_source().node.soft_get('type') == 'Const':
+                return
+        # 4. check that offset of begin and end values for slicing axis is constant
+        add_node = end_node.in_port(sliced_axis).get_source().node
+        slice_start_index_node = begin_node.in_port(sliced_axis).get_source().node
+        if add_node.soft_get('type') != 'Add':
+            return
+        slice_size_node = add_node.in_port(1).get_source().node
+        if slice_size_node.soft_get('type') != 'Const':
+            return
+        slice_size = slice_size_node.soft_get('value')
+        if add_node.in_port(0).get_source().node.id != slice_start_index_node.id:
+            return
+        step_value = step_node.soft_get('value')[sliced_axis]
+        # 5. check that step_value equal to 1 and step_value equal to 1
+        # TODO: support other cases when slice_size not equal to 1 and step_value not equal to 1
+        if slice_size != 1 or step_value != 1:
```

Comment:
I have already thought of the case when `slice_size != 1 or step_value != 1` but generalization will lead to complicated transformation. It will use `Range` operation to generate a range of indices by which to slice and also it will require to change new TensorIteratorInput transformation. So bigger extension is required. So I postpone this case for a while.